depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp

```diff
@@ -150,19 +150,23 @@ class DeviceROS : public dai::DeviceBase {
 
 protected:
     // create stream based on name at the time of subscription
-    template <class MsgType, dai::DatatypeEnum DataType>
-    auto generate_cb_lambda(std::unique_ptr<dai::XLinkStream>& stream, msgpack::sbuffer& sbuf,
-            std::vector<std::uint8_t>& writer_buf) -> boost::function<void(const boost::shared_ptr<MsgType const>&)> {
-        const auto core_sub_lambda = [&stream, &sbuf, &writer_buf](const boost::shared_ptr<MsgType const>& msg) {
+    template <class MsgType>
+    auto generate_cb_lambda(const std::string& name) -> boost::function<void(const boost::shared_ptr<MsgType const>&)> {
+                std::unique_ptr<dai::XLinkStream>& stream = _streams[name];
+                std::vector<std::uint8_t>& serial_buf = _serial_bufs[name];
+                std::vector<std::uint8_t>& writer_buf = _writer_bufs[name];
+
+        const auto core_sub_lambda = [&stream, &serial_buf, &writer_buf](const boost::shared_ptr<MsgType const>& msg) {
```

Comment:
Would stream be a reference to _streams[name] or a reference to stream on L155?

Reply:
what does this mean?

Reply:
I wonder if we have a local reference which might dangle

Reply:
the reference becomes dangling as you points out.
https://try2explore.com/questions/10597846

I will fix it.

Reply:
this is working, but is this a good way?
```
        auto* stream = &_streams[name];
        auto* serial_buf = &_serial_bufs[name];
        auto* writer_buf = &_writer_bufs[name];

        const auto core_sub_lambda = [stream, serial_buf, writer_buf](const boost::shared_ptr<MsgType const>& msg) {
            Guard guard([] { ROS_ERROR("Communication failed: Device error or misconfiguration."); });
```

Reply:
You can move the capture directly to the lambda capture space?

Reply:
I tried that by 
```
        const auto core_sub_lambda = [stream = std::move(_streams[name]), &serial_buf, &writer_buf](const boost::shared_ptr<MsgType const>& msg) {
```
but that could not compile by 
```
In file included from /home/kota/ws_depthai/src/depthai_ros/depthai_ros_driver/src/dai_utils.cpp:2:0:
/home/kota/ws_depthai/src/depthai_ros/depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp: In instantiation of ‘boost::function<void(const boost::shared_ptr<const M>&)> rr::DeviceROS::generate_cb_lambda(const string&) [with MsgType = depthai_datatype_msgs::RawCameraControl_<std::allocator<void> >; std::__cxx11::string = std::__cxx11::basic_string<char>]’:
/home/kota/ws_depthai/src/depthai_ros/depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:306:139:   required from here
/home/kota/ws_depthai/src/depthai_ros/depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:178:16: error: use of deleted function ‘rr::DeviceROS::generate_cb_lambda(const string&) [with MsgType = depthai_datatype_msgs::RawCameraControl_<std::allocator<void> >; std::__cxx11::string = std::__cxx11::basic_string<char>]::<lambda(const boost::shared_ptr<const depthai_datatype_msgs::RawCameraControl_<std::allocator<void> > >&)>::<lambda>(const rr::DeviceROS::generate_cb_lambda(const string&) [with MsgType = depthai_datatype_msgs::RawCameraControl_<std::allocator<void> >; std::__cxx11::string = std::__cxx11::basic_string<char>]::<lambda(const boost::shared_ptr<const depthai_datatype_msgs::RawCameraControl_<std::allocator<void> > >&)>&)’
         return core_sub_lambda;
```

Reply:
`serial_buf` and `writer_buf` could be moved to the capture with `mutable`, but `stream` could not be moved...

Reply:
Hmm,  this can compile
```
const auto core_sub_lambda = [&stream = _streams[name], &serial_buf, &writer_buf](const boost::shared_ptr<MsgType const>& msg) {
```